fastlane/Fastfile

```diff
@@ -0,0 +1,61 @@
+opt_out_usage
+default_platform :mac
+
+desc "Increment (with 'bump' option) or set (with 'pre' option) the framework version"
+lane :bump_version do |options|
+  version = version_bump_xcconfig(path: "Configurations/Base.xcconfig", bump_type: options[:bump] && options[:bump].to_sym, version_pre: "beta.#{options[:beta]}")
+  deferred_readme_set_version(path: "Documentation/Guide/Getting Started.md", version: version)
+  podspec_set_version(path: "BNRDeferred.podspec", version: version)
+  puts version
+end
+
+desc "Run CocoaPods linter"
+lane :pod_lint do
+  pod_lib_lint(quick: true)
+end
+
+desc "Output documentation using Jazzy into docs/"
+lane :build_docs do
+  jazzy(config: ".jazzy.yml")
+end
+
+desc "Build and publish documentation from docs/ into gh-pages"
+lane :publish_docs do
+  publish_gh_pages(path: "docs", author_name: "Big Nerd", author_email: "cocoa-engineering@bignerdranch.com")
+end
+
+platform :mac do
+  desc "Test using Swift Package Manager for macOS and Linux"
+  lane :test do
+    sh "swift test"
+  end
+
+  desc "Execute tests, perform CocoaPods linting, publish documentation"
+  lane :ci do
+    test
+    pod_lint
```

Comment:
I did some of this, but probably not how you intended. I am adverse to mucking more with Travis matrix stuff (because I don't like them), and I welcome improvements here otherwise.